django_socio_grpc/proto_serializers.py

```diff
@@ -76,59 +85,59 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
-                    continue
-
-                # INFO - AM - 11/03/2024 - Here we set the default value especially for the blank authorized data. We debated about raising a ValidaitonError but prefered this behavior. Can be changed if it create issue with users
-                data_dict[field.field_name] = message.DESCRIPTOR.fields_by_name[
-                    field.field_name
-                ].default_value
-
-            if field.allow_null or (field.default in [None, empty] and field.required is True):
-                if is_update_process:
-                    data_dict[field.field_name] = None
-                    continue
-
-                if field.default not in [None, empty]:
-                    data_dict[field.field_name] = None
-                    continue
-
-                if (
-                    hasattr(self, "Meta")
-                    and hasattr(self.Meta, "model")
-                    and hasattr(self.Meta.model, field.field_name)
-                ):
-                    deferred_attribute = getattr(self.Meta.model, field.field_name)
-                    if deferred_attribute.field.default != NOT_PROVIDED:
-                        data_dict[field.field_name] = get_default_value(
-                            deferred_attribute.field.default
-                        )
-                        continue
-
-                data_dict[field.field_name] = None
-        return data_dict
+            with contextlib.suppress(_NoDictData):
+                clean_dict[field.field_name] = self._get_cleaned_data(
+                    partial_fields, field, is_update_process, message
+                )
+        return clean_dict
+
+    def _get_cleaned_data(
+        self,
+        partial_fields: List[str],
+        field: Field,
+        is_update_process: bool,
+        message,
+    ):
+        # INFO - AM - 04/01/2024 - if field is not in the data_dict but in PARTIAL_UPDATE_FIELD_NAME
+        # we need to set the default value if existing or raise exception to avoid having default
+        # grpc value by mistake
+        if field.field_name in partial_fields:
+            if field.allow_null:
+                return None
+            elif field.default not in [None, empty]:
+                return get_default_value(field.default)
+            # INFO - AM - 11/03/2024 - Here we set the default value especially for the blank authorized data.
+            # We debated about raising a ValidaitonError but prefered this behavior.
+            # Can be changed if it create issue with users
+            return message.DESCRIPTOR.fields_by_name[field.field_name].default_value
+
+        elif field.allow_null or (field.default in [None, empty] and field.required is True):
+            if is_update_process or field.default not in [None, empty]:
+                return None
+            try:
+                deferred_attribute = getattr(self.Meta.model, field.field_name)
```

Comment:
Meta and model in meta can be not set no ?

Reply:
that's why it's in a try except